delay/src/lib.rs

```diff
@@ -267,41 +310,55 @@ impl Waiter for ThrottleWaiter {
 
 #[derive(Clone)]
 struct ExponentialBackoffWaiter {
-    next: RefCell<Duration>,
+    next: Option<RefCell<Duration>>,
     initial: Duration,
     multiplier: f32,
     cap: Duration,
 }
 impl ExponentialBackoffWaiter {
     pub fn new(initial: Duration, multiplier: f32, cap: Duration) -> Self {
         ExponentialBackoffWaiter {
-            next: RefCell::new(initial),
+            next: None,
             initial,
             multiplier,
             cap,
         }
     }
 }
 impl Waiter for ExponentialBackoffWaiter {
+    fn restart(&mut self) -> Result<(), WaiterError> {
+        match &self.next {
+            Some(next) => {
+                next.replace(self.initial);
+                Ok(())
+            }
+            None => Err(WaiterError::NotStarted),
+        }
+    }
+
     fn start(&mut self) {
-        self.next = RefCell::new(self.initial);
+        self.next = Some(RefCell::new(self.initial));
     }
 
     fn wait(&self) -> Result<(), WaiterError> {
-        let current = *self.next.borrow();
-        let current_nsec = current.as_nanos() as f32;
+        if let Some(next) = &self.next {
```

Comment:
Can you simplify this with `let next = &self.next.ok_or(WaiterError::NotStarted)?;`?

Reply:
Done!